UPGRADE.md

```diff
@@ -8,6 +8,11 @@ awareness about deprecated code.
 
 # Upgrade to 4.0
 
+## BC BREAK: Types can no longer be reverse-engineered reliably
```

Comment:
Reverse engineering is a really odd term in this context. It implies that there's no source of the data being obtained. In most cases, there is a source. I wouldn't use "reliably" either. Reliability is about something working or not working depending on a multitude of factors, usually probabilistic, while here it depends only on the column type.

I'd put it as something like "Deployed database schema no longer contains the information about abstract data types".